src/ciphers/caesar.rs

```diff
@@ -1,20 +1,25 @@
-//! Caesar Cipher
-//! Based on cipher_crypt::caesar
-//!
-//! # Algorithm
-//!
-//! Rotate each ascii character by shift. The most basic example is ROT 13, which rotates 'a' to
-//! 'n'. This implementation does not rotate unicode characters.
-
-/// Caesar cipher to rotate cipher text by shift and return an owned String.
-pub fn caesar(cipher: &str, shift: u8) -> String {
-    cipher
-        .chars()
+/// Encrypts a given text using the Caesar cipher technique.
+///
+/// In cryptography, a Caesar cipher, also known as Caesar's cipher, the shift cipher, Caesar's code,
+/// or Caesar shift, is one of the simplest and most widely known encryption techniques.
+/// It is a type of substitution cipher in which each letter in the plaintext is replaced by a letter
+/// some fixed number of positions down the alphabet.
+///
+/// # Arguments
+///
+/// * `text` - The text to be encrypted.
+/// * `rotation` - The number of rotations (shift) to be applied.
+///
+/// # Returns
+///
+/// Returns the encrypted string.
+pub fn caesar(text: &str, rotation: usize) -> String {
```

Comment:
I would check if `0 <= rotation < 26`, because using any other rotation does not make too much sense.

Reply:
That makes sense. What do you think about returning an error if the rotation is not within the range (0 - 25), changing the return type to `Result<String, &'static str>`?

Reply:
Go for it. Maybe even without explicitly defining `ERROR_MESSAGE`.